execution/executor/tests/storage_integration_test.rs

```diff
@@ -247,6 +247,520 @@ fn test_reconfiguration() {
     // validator set fails to parse
 }
 
+#[test]
+fn test_change_publishing_option_to_custom() {
+    // Publishing Option is set to locked at genesis.
+    let mut rt = Runtime::new().unwrap();
+    let (mut config, genesis_key) = config_builder::test_config();
+
+    let (_storage_server_handle, executor, mut committed_trees) =
+        create_storage_service_and_executor(&config);
+    let storage_read_client = Arc::new(StorageReadServiceClient::new(&config.storage.address));
+
+    let genesis_account = association_address();
+    let network_config = config.validator_network.as_ref().unwrap();
+    let validator_account = network_config.peer_id;
+    let keys = config
+        .test
+        .as_mut()
+        .unwrap()
+        .account_keypair
+        .as_mut()
+        .unwrap();
+
+    let validator_privkey = keys.take_private().unwrap();
+    let validator_pubkey = keys.public().clone();
+    let auth_key = AuthenticationKey::ed25519(&validator_pubkey);
+    let validator_auth_key_prefix = auth_key.prefix().to_vec();
+    assert!(
+        auth_key.derived_address() == validator_account,
+        "Address derived from validator auth key does not match validator account address"
+    );
+
+    // give the validator some money so they can send a tx
+    let txn1 = get_test_signed_transaction(
+        genesis_account,
+        /* sequence_number = */ 1,
+        genesis_key.clone(),
+        genesis_key.public_key(),
+        Some(encode_transfer_script(
+            &validator_account,
+            validator_auth_key_prefix,
+            1_000_000,
+        )),
+    );
+
+    let script1 = Script::new(vec![], vec![]);
+    let script2 = Script::new(vec![1], vec![]);
+
+    // Create a transaction that is not allowed with default publishing option and make sure it is
+    // rejected.
+    let txn2 = get_test_signed_transaction(
+        validator_account,
+        /* sequence_number = */ 0,
+        validator_privkey.clone(),
+        validator_pubkey.clone(),
+        Some(script1.clone()),
+    );
+
+    let txn3 = get_test_signed_transaction(
+        validator_account,
+        /* sequence_number = */ 0,
+        validator_privkey.clone(),
+        validator_pubkey.clone(),
+        Some(script2.clone()),
+    );
+
+    // Create a dummy block prologue transaction that will bump the timer.
```

Comment:
Why is this needed?

Reply:
The executor is spawn up from the genesis (with a reconfiguration event in it). We need the timer to be bumped in order to trigger a new reconfiguration event!